stack.yaml

```diff
@@ -1,25 +1,28 @@
-resolver: lts-8.15
+resolver: lts-12.16
 
 packages:
 - '.'
 
 extra-deps:
-- git: https://github.com/dapphub/hevm.git
-  commit: 40b4ce9d38ed711ffd6065e697807100b3f0a3c3
-- brick-0.18
-- config-ini-0.2.1.1
-- data-clist-0.1.2.0
 - ghci-pretty-0.0.2
-- hedgehog-0.6
 - HSH-2.1.3
 - ipprint-0.6
-- megaparsec-6.4.0
-- optparse-applicative-0.13.2.0
-- parser-combinators-0.4.0
-- restless-git-0.5.0
+- restless-git-0.7
 - rosezipper-0.2
 - sr-extra-1.46.3.2
 - temporary-1.2.1.1
 - tree-view-0.5
-- vty-5.16
-- word-wrap-0.4.1
+- s-cargot-0.1.4.0
+- Unixutils-1.54.1
+- readline-1.0.3.0
+- hevm-0.21
+- multiset-0.3.4.1
+- text-format-0.3.2
+
+nix:
```

Comment:
I don't understand why this section section is necessary if the travis installation does not require NIX. Please remove this if it is useless. Also, update the README to make sure the instructions are correct.

Reply:
It is not necessary for CI, but it is necessary if someone wants to use stack's nix integration. It let's someone build echidna using `stack build --nix` instead of relying on system libraries that may or may not exist. In the case of libsecp256k1, it almost certainly would not exist. This list of packages also serves as documentation of what system deps echidna needs. I think we should keep this here so I can put `stack build --nix` as one option for building (I will also make sure that the steps for building without nix are documented as well). Does this sound reasonable to you?

Reply:
bump

Reply:
While it sounds reasonable to me, we are not currently supporting nix to build echidna. We would love to do it, but we a have a few reasons to avoid it. If we decided to accept this change, we *should* keep updated this list of packages, in case someone want to use it. But since we are not using nix, that's why I asked you to delete that section. If you want, feel free to support an installation using nix in your own branch. We could even link your branch from our README, in case someone is interested in using nix.

Reply:
What's the advantage of using `stack --nix` rather than just adding a nix derivation for echidna to nixpkgs?

Also, IIRC all hackage packages are automatically built on nixpkgs.

Reply:
@andremarianiello once echidna is on hackage (the Haskell package repository) it **should** be picked up within a week or so by the nix infrastructure automatically.

There might be build issues and then we'd have to look into it, but I think pushing to Hackage is the first step.

Reply:
But now I realize that the above tool only works with packages using Cabal, not Stack. There's [stack2nix](https://github.com/input-output-hk/stack2nix) for that, but the last time I tried packaging echidna with it, it was broken. :man_shrugging:

Reply:
@ggrieco-tob Can expand on

> we are not currently supporting nix to build echidna. We would love to do it, but we a have a few reasons to avoid it.